### ISSUE TITLE
Agree on wiring convention for spinful fermion systems

### ISSUE THREAD

**Issue by LaurinFischer** [Qiskit-Extensions/qiskit-cold-atom#28]
The fermionic systems in Qiskit-cold-atom support spinful fermions. 

Currently, when a spin is chosen (e.g. s=1/2), the quantum circuit will consist of multiple registers which describe the spinful fermionic modes. For each spin value, one quantum register is created, which are then put together to form the quantum circuit. 
Here's an example:

```python
provider = ColdAtomProvider()
backend = provider.get_backend("fermionic_tweezer_simulator")
qc = backend.initialize_circuit([[1, 0, 0, 1], [0, 0, 1, 1]])

qc.qregs
>>> [QuantumRegister(4, 'spin_0'), QuantumRegister(4, 'spin_1')]
```
I think that this separation of different spin components into different quantum registers makes sense and it is the natural way to structure circuits in Qiskit. It also makes it easy to set up a device with more than spin-1/2, e.g. a fermionic system with three different spin species.
Unfortunately, there is no easy way of rearranging the wires of the various quantum registers to get an "interleafed" circuit, where the spin species alternate. This convention is used by Fred's simulators and the circuit description they used for pennylane. 

### How to proceed ?

I think an argument can be made for the following options:
1. Stick with what we have in Qiskit. In this case the user has to figure out which convention is used by which backend, which is a bit cumbersome. We could add a flag for this in the backend configuration. 
2. Adapt the qsimsim to the Qiskit notation (don't know if this would break the pennylane support).
3. Change to the interleaved convention in Qiskit by just using one global quantum register. 
4. Change to the interleaved convention in Qiskit. Structure the circuit not by the spin species but by the spatial mode, so something like this:
```python
# example for a 4-site, 2-spin circuit
qc = backend.initialize_circuit([[1, 0], [0, 0], [0, 1], [1, 1]])

qc.qregs
>>> [QuantumRegister(2, 'site_0'), QuantumRegister(2, 'site_1'), QuantumRegister(2, 'site_2'), QuantumRegister(2, 'site_3')]
```

3. and 4. would require some careful refactoring of the fermionic systems. Doable, but I don't think I would get to this in the next 2 weeks.

**Comment by fretchen** [comment]
Hi,

my preferred option is 1) but let me explain why.

3. and 4. seem to adopt a new notation which is not really compatible with the rest of qiskit. In other words, I assume that you chose your notation to be compatible with `qiskit-nature`. Then you should totally never ever break this IMHO.
2. Actually really breaks the connection to `pennylane` and it does not solve the issue that someone will come back claiming that qiskit did it all wrong.
1. This is actually kind of nice. Ideally the back-end only needs to know that `qiskit` works with this notation of all spin-up before all spin-down. This would mean that the json that is sent to the back-end contains a boolean that is called something like `interleafed` and then the back-end can handle all the opinions out there. It makes it possible to have consistent notations within each programming framework. And it allows each back-end to chose its preferred ordering ...

**Comment by eggerdj** [comment]
Here's my take on this so far. I think 1) is a nice option. We preserve what we have in Qiskit. In the job-payload we provide a flag, e.g. `wire_order` which in Qiskit's case would look like `"wire_order": "consecutive"` (the other option being `"interleaved"`). Then the backend can deal with this how it pleases. Do we want the backend to report it's preferred order in the configuration? Seems nice but then we would just ignore it. Or at least what it will assume by default?

Looking at the code we would need to adapt `circuit_to_cold_atom.py` with the extra field. E.g.
```python
    experiments = {}
    for idx, circuit in enumerate(circuits):
        experiments["experiment_%i" % idx] = {
            "instructions": circuit_to_data(circuit),
            "shots": shots,
            "num_wires": circuit.num_qubits,
            "wire_order": "consecutive",
        }

    return experiments
```
Hard-coding this in the dict is not so nice. We could make this Qiskit choice more prominent in the code. Perhaps by putting `circuit_to_cold_atom` and `circuit_to_data` in a class and making the wire order a class variable.

From a Qiskit perspective this is all easy since it moves this complexity to the backend.

**Comment by fretchen** [comment]
I like the proposed approach with the "wire_order". We will implement it, when we have time...

**Comment by eggerdj** [comment]
Solved by #31
